feat(empty-state): DLT-2361 DLT-2362 update icon color

jcarter-dp opened the pull request:
# feat(empty-state): Update icon color

## :hammer_and_wrench: Type Of Change

- [x] Feature

## :book: Jira Ticket

<!--- Enter the URL of the Jira ticket associated with this PR -->

https://dialpad.atlassian.net/browse/DLT-2361
https://dialpad.atlassian.net/browse/DLT-2362

## :book: Description

<!--- Describe specifically what the changes are -->
Reviewed group 2 and group 3 of css variables and making sure they match within the figma rebrand-2025 file. All looked good, only updated css variable was the empty state's icon which is now set to "muted"

## :pencil: Checklist

<!--- Tick or place an `x` in all of the checkboxes that apply -->
<!--- Remove checkboxes that do not apply -->

For all PRs:

- [x] I have ensured no private Dialpad links or info are in the code or pull request description (Dialtone is a public repo!).
- [ ] I have reviewed my changes.
- [ ] I have added all relevant documentation.
- [ ] I have considered the performance impact of my change.


For all CSS changes:

- [x] I have used design tokens whenever possible.
- [x] I have considered how this change will behave on different screen sizes.
- [x] I have visually validated my change in light and dark mode.
- [x] I have used gap or flexbox properties for layout instead of margin whenever possible.

josedialpad commented on docs/_data/card.json:
```diff
@@ -0,0 +1,24 @@
+    },
+    {
+      "class": "d-card__header",
+      "applies": "Child of <span class=\"code-example--inline\">.d-card</span>",
```
Not specific to this PR, but noticed it's not rendering the markup when it also has plain text
![2022-08-31 at 17 13 55@2x](https://user-images.githubusercontent.com/83774467/187773509-8dd748ac-cf79-4112-91cf-7aaa20c35e93.png)

josedialpad replied:
Created a ticket for this https://dialpad.atlassian.net/browse/DT-649. So it's not needed to fix in this PR.

josedialpad commented on docs/components/card.md:
```diff
@@ -0,0 +1,130 @@
+title: Card
+desc: A card contains summary content and actions about a single subject. It can be used by itself or within a list, and is generally interactive.
+storybook_url: https://vue.dialpad.design/?path=/story/components-card--default
+prev:
+text: Utilities - Word break
+link: /utilities/typography/word-break
```
```suggestion
```
unnecessary?

shefaligupta replied:
copy paste issues! :D

josedialpad commented on docs/components/card.md:
```diff
@@ -0,0 +1,130 @@
+  </div>
+</code-well-header>
+
+[//]: # (## Usage)
+[//]: # (Lorem ipsum dolor sit amet, consectetur adipiscing elit. Morbi massa ante, tempus vitae lacus id, luctus tristique lorem. Mauris feugiat massa ex, id aliquet mi tempor non. Curabitur non tristique lectus. Fusce ut nisl non diam dignissim viverra. In posuere dui arcu, sed eleifend massa faucibus sed. Phasellus quis leo vitae erat pellentesque venenatis id vitae lectus. Suspendisse convall ...
```
```suggestion
```

josedialpad reviewed (commented):
Good work!

I think we can add some content for the usage and a11y, for instance, some ideas for the usage section:

```
Use for: 
- To display content and actions on a single topic.

Don't use for:
- Add too many call-to-action elements to the same card. A card should only contain a single primary action.
- Inform users about important changes.

Best Practices
- It should only contain a single idea that may feature a call-to-action, or the option to navigate to more detailed content.
- The content of a card should be concise and offer only a preview of detailed content.
- The headings should set clear expectations about the card’s purpose.
```

shefaligupta commented:
> Good work!
> 
> I think we can add some content for the usage and a11y, for instance, some ideas for the usage section:
> 
> ```
> Use for: 
> - To display content and actions on a single topic.
> 
> Don't use for:
> - Add too many call-to-action elements to the same card. A card should only contain a single primary action.
> - Inform users about important changes.
> 
> Best Practices
> - It should only contain a single idea that may feature a call-to-action, or the option to navigate to more detailed content.
> - The content of a card should be concise and offer only a preview of detailed content.
> - The headings should set clear expectations about the card’s purpose.
> ```

This is what I was looking for. You are the best @josedialpad :)

josedialpad commented:
Is there any figma or design for this component?

shefaligupta commented:
> Is there any figma or design for this component?

This is what @dialpad-drew put for me initially - https://www.figma.com/file/ttP4nkAOdjspCw4DvrQJwR/Card-Proposal

braddialpad commented on docs/components/card.md:
```diff
@@ -0,0 +1,157 @@
+    <button href="#" class="d-btn d-btn--xs d-fc-black-400"><IconMenuVertical /></button>
+  </div>
+  <div class="d-card__content d-h72">
+    <div class="d-h12 d-w84 d-bgc-black-400 d-mb8" />
+    <div class="d-h12 d-w216 d-bgc-black-400 d-mb8" />
+    <div class="d-h12 d-w164 d-bgc-black-400" />
```
This was just a placeholder to represent text used in the design right? Do you think we should use textual content instead on the actual dialtone doc site since this better represents actual usage (a few lines of lorem ipsum would be fine)

shefaligupta replied:
<img width="976" alt="Screen Shot 2022-08-31 at 4 48 03 PM" src="https://user-images.githubusercontent.com/7941880/187798333-1e5d3b55-d188-4390-be0b-294ff4ee6762.png">

How about this? @braddialpad

braddialpad replied:
Looks fine, but you probably want to use actual generated lorem ipsum instead of a description of what it is 😄.

something like this:

```html
<p class="d-fs14 d-fw-bold d-fc-black-800">Lorem ipsum</p>
<p class="d-fs12">Lorem ipsum dolor sit amet, consectetur adipiscing elit. Donec fermentum molestie semper. Morbi finibus nulla turpis, nec molestie mi rutrum.</p>
```

braddialpad replied:
looks good!

braddialpad commented on lib/build/less/components/card.less:
```diff
@@ -0,0 +1,72 @@
+  align-items: center;
+  justify-content: space-between;
+  width: 100%;
+  margin: 0 !important;
```
Do we need important here?

braddialpad reviewed (commented):
Thanks Shefali, excited to finally have this!

josedialpad reviewed (approved):
LGTM, a final question for me is if we want to show the Lorem ipsum text in the other variants too.

<img width="983" alt="2022-09-01 at 11 45 39@2x" src="https://user-images.githubusercontent.com/83774467/187943505-0ba29769-0aa5-4b4c-a024-228ea7967326.png">

shefaligupta commented:
> LGTM, a final question for me is if we want to show the Lorem ipsum text in the other variants too.
> 
> 
> 
> <img width="983" alt="2022-09-01 at 11 45 39@2x" src="https://user-images.githubusercontent.com/83774467/187943505-0ba29769-0aa5-4b4c-a024-228ea7967326.png">

@braddialpad should I change the variants too? Or is it good as is?

braddialpad commented:
> @braddialpad should I change the variants too? Or is it good as is?

Yeah change the other variants as well

braddialpad commented:
> Good work!
> 
> I think we can add some content for the usage and a11y, for instance, some ideas for the usage section:
> 
> ```
> Use for: 
> - To display content and actions on a single topic.
> 
> Don't use for:
> - Add too many call-to-action elements to the same card. A card should only contain a single primary action.
> - Inform users about important changes.
> 
> Best Practices
> - It should only contain a single idea that may feature a call-to-action, or the option to navigate to more detailed content.
> - The content of a card should be concise and offer only a preview of detailed content.
> - The headings should set clear expectations about the card’s purpose.
> ```

@josedialpad did you mean Shefali should add this now? Or we will add later?

shefaligupta commented:
> > Good work!
> > I think we can add some content for the usage and a11y, for instance, some ideas for the usage section:
> > ```
> > Use for: 
> > - To display content and actions on a single topic.
> > 
> > Don't use for:
> > - Add too many call-to-action elements to the same card. A card should only contain a single primary action.
> > - Inform users about important changes.
> > 
> > Best Practices
> > - It should only contain a single idea that may feature a call-to-action, or the option to navigate to more detailed content.
> > - The content of a card should be concise and offer only a preview of detailed content.
> > - The headings should set clear expectations about the card’s purpose.
> > ```
> 
> @josedialpad did you mean Shefali should add this now? Or we will add later?

oh I have added this already.

braddialpad commented:
> oh I have added this already.

My mistake, yep looks good.

shefaligupta commented:
@braddialpad - added lorem ipsum for the variants. Can you please verify the latest preview - made some padding changes per Drew's card proposal FIGMA.

braddialpad reviewed (approved):
Good by me, thanks!

jcarter-dp committed:
feat(design-tokens): Apply new color to DT Code components: Group 2

Update and review CSS to reflect color changes made in the Figma Components within group 2 and group 3.

francisrupert reviewed (approved):
Since this only wound up affecting Empty State, change the PR name to `feat(empty-state): DLT-2361 DLT-2362 update icon color` before merging. 

That way the release notes are more specific to the change, which is effectively independent of the new colors. I say that because that icon color was always meant to be `muted`, but Fede first had to correct some icons to play better with overlapping icon strokes.

Change the PR name and you're good to merge!
